neuralop/layers/spectral_convolution.py

```diff
@@ -493,7 +493,34 @@ def forward(
 
         if self.order > 1:
             out_fft = torch.fft.fftshift(out_fft, dim=fft_dims[:-1])
+
+        # not all CUFFT backend algorithms ensure that the imaginary components 
+        # of zero and nyquist (n//2) frequencies are set to 0, so we do this manually
+
+        zero_indices = []
+        nyquist_indices = []
+        for i, mode_sz in enumerate(fft_size):
+            # b, c, *spatial_dims
+
+            # after reversing fft shift, zero is at 0 
+            zero_indices.append([slice(None), slice(None)] + [slice(None)] * i +\
+                                        [slice(0, 1)] + [slice(None)] * (len(fft_size) - i - 1))
+            
+        # after reversing fft shift, nyquist is at n//2 
+        for i, mode_sz in enumerate(fft_size[:-1]):
+            if mode_sz % 2 == 0:
+                nyquist_indices.append([slice(None), slice(None)] + [slice(None)] * i +\
+                                        [slice(mode_sz//2, mode_sz//2 + 1)] + [slice(None)] * (len(fft_size) - i - 1))
+        # except for the last mode, where it is at 
+        nyquist_indices.append((slice(None), slice(None), *[slice(None)]*(len(fft_size)-1), slice(-1)))
+            
+        for idx in zero_indices:
+            out_fft[idx].imag = 0.0
         
```

Comment:
this looks correct to me but I would double-check by running some code on a power of 2 example and making sure that this fix removed the artifact

Reply:
No artifact appears